rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -861,107 +859,79 @@ rmw_ret_t configure_qos_for_security(
+  this->domain_id = static_cast<dds_domainid_t>(options->domain_id);
+  if (!check_create_domain(this->domain_id, options->localhost_only)) {
     return RMW_RET_ERROR;
   }
 
-  /* Once the domain id is set in impl, impl's destructor will take care of unref'ing
-     the domain */
-  impl->domain_id = domain_id;
-
   std::unique_ptr<dds_qos_t, std::function<void(dds_qos_t *)>>
   ppant_qos(dds_create_qos(), &dds_delete_qos);
   if (ppant_qos == nullptr) {
     return RMW_RET_BAD_ALLOC;
   }
   std::string user_data = std::string("enclave=") + std::string(
-    context->options.enclave) + std::string(";");
+    options->enclave) + std::string(";");
   dds_qset_userdata(ppant_qos.get(), user_data.c_str(), user_data.size());
   if (configure_qos_for_security(
       ppant_qos.get(),
-      &context->options.security_options) != RMW_RET_OK)
+      &options->security_options) != RMW_RET_OK)
   {
-    if (context->options.security_options.enforce_security == RMW_SECURITY_ENFORCEMENT_ENFORCE) {
+    if (RMW_SECURITY_ENFORCEMENT_ENFORCE == options->security_options.enforce_security) {
       return RMW_RET_ERROR;
     }
   }
-  impl->ppant = dds_create_participant(domain_id, ppant_qos.get(), nullptr);
-  if (impl->ppant < 0) {
+
+  this->ppant = dds_create_participant(this->domain_id, ppant_qos.get(), nullptr);
+  if (this->ppant < 0) {
     RCUTILS_LOG_ERROR_NAMED(
       "rmw_cyclonedds_cpp", "rmw_create_node: failed to create DDS participant");
     return RMW_RET_ERROR;
   }
 
   /* Create readers for DDS built-in topics for monitoring discovery */
-  if ((impl->rd_participant =
-    dds_create_reader(impl->ppant, DDS_BUILTIN_TOPIC_DCPSPARTICIPANT, nullptr, nullptr)) < 0)
+  if ((this->rd_participant =
+    dds_create_reader(this->ppant, DDS_BUILTIN_TOPIC_DCPSPARTICIPANT, nullptr, nullptr)) < 0)
   {
     RCUTILS_LOG_ERROR_NAMED(
       "rmw_cyclonedds_cpp", "rmw_create_node: failed to create DCPSParticipant reader");
     return RMW_RET_ERROR;
   }
-  if ((impl->rd_subscription =
-    dds_create_reader(impl->ppant, DDS_BUILTIN_TOPIC_DCPSSUBSCRIPTION, nullptr, nullptr)) < 0)
+  if ((this->rd_subscription =
+    dds_create_reader(this->ppant, DDS_BUILTIN_TOPIC_DCPSSUBSCRIPTION, nullptr, nullptr)) < 0)
   {
     RCUTILS_LOG_ERROR_NAMED(
       "rmw_cyclonedds_cpp", "rmw_create_node: failed to create DCPSSubscription reader");
     return RMW_RET_ERROR;
   }
-  if ((impl->rd_publication =
-    dds_create_reader(impl->ppant, DDS_BUILTIN_TOPIC_DCPSPUBLICATION, nullptr, nullptr)) < 0)
+  if ((this->rd_publication =
+    dds_create_reader(this->ppant, DDS_BUILTIN_TOPIC_DCPSPUBLICATION, nullptr, nullptr)) < 0)
   {
     RCUTILS_LOG_ERROR_NAMED(
       "rmw_cyclonedds_cpp", "rmw_create_node: failed to create DCPSPublication reader");
     return RMW_RET_ERROR;
   }
 
   /* Create DDS publisher/subscriber objects that will be used for all DDS writers/readers
-     to be created for RMW publishers/subscriptions. */
```

Comment:
nit, accidental whitespace change?

Reply:
Solved in https://github.com/ros2/rmw_cyclonedds/pull/176/commits/ded89f9ab699905649468ff68e128f3587f9c09d.